tests/testthat/test-rownames.R

```diff
@@ -26,6 +26,13 @@ test_that("rownames_to_column keeps the tbl classes (#882)", {
                paste("There is a column named wt already!")  )
 })
 
+test_that("rowid_to_column adds row names and errors if col name reused", {
+  res <- rowid_to_column( mtcars)
```

Comment:
I think you can adapt most of the tests for `rownames_to_column()` to the new function; in particular it seems important to test that the class is maintained (i.e., that it also works for data frames).

Unfortunately, that test isn't a particularly good example for good coding style; I prefer spaces around operators but not before/after parens, and an indent of two lines for function calls that occupy multiple lines, with nothing after the opening or before the closing paren. The full style guide is at http://style.tidyverse.org .